### PR TITLE
Issue #2975 - domain and domain info portfolio fields -[HOTGOV]

### LINKED ISSUE #2975: Domain Info and domain on Django admin pull portfolio info when applicable
### Issue description

For org model Domains/ Domain info, for all portfolio specific fields we should be pulling from the portfolio instead of the domain info object. For example, organization name should come from the portfolio, as changes should be done on the portfolio not on domain info. However, we only want to perform these changes on the django admin tables, all exports (domain exports, analytics pages exports), and any analytics graphs where this applicable. 

### Acceptance criteria

- [ ] Follow what was done in #2852 
- [ ] Domain Info/Domain fields on django admin should be have a corresponding function that is returning either the stored domain info value or the corresponding portfolio value where appropriate. 
- [ ] Filters, sorts and table views on domain info &domain on Django admin are all updated to now pull from the portfolio when needed
- [ ] All reports are updated to use the appropriate fields (this includes those on the analytics page and on the tables) 
- [ ] Double check all domain request reports are also using similar logic and update accordingly
- [ ] Do not change the change form (domain/domain info record) view

### Additional context
implementation suggestion note: we should be have a corresponding @property function  for each domain info field that is used for reports/analytics/ django admin table/django admin filter. Below are  Domain Info/Domain fields but you will not need to handle each one, only create @property fields **IF** we are using this value for a report/ analytics board or to display on a table:
  - [ ]  organization_name
  - [ ]  generic_org_type  (called "organization_type" in Portfolio)
  - [ ]  federal_agency - should come from federal agency from federal agency entry associated with the portfolio
  - [ ]  federal_type - should come from federal agency table entry associated with the portfolio
  - [ ]  senior_official -should come from portfolio
  - [ ]  address_line1 - should be from portfolio organization address
  - [ ]  address_line2 - should be from portfolio organization address
  - [ ]  state_territory - should be from portfolio organization address
  - [ ]  zipcode - should be from portfolio organization address
  - [ ]  urbanization - should be from portfolio organization address

### Links to other issues

- blocked by #2852 
- blocks #2867

### PR DESCRIPTION
**Pull request by CocoByte**
## Ticket 2975

Resolves #2975

## Changes

- Added @property functions to DomainInfo to pull data from Portfolio when a portfolio is present
- Updated admin.py so that DomainInfo and Domain tables in /admin use the converted fields in their columns and filters
- Updated csv exports for the following;
     - Domain, Domain Info, and Domain Request admin tables
     - All analytic tables
- Updated unit tests to use converted fields

## Code Review Verification Steps

Go to /admin and verify that the following csv exports use converted fields instead of the native data for Domains, Domain Requests, and Domain Info;

- [ ] Domains table
- [ ] Domain Requests table
- [ ] Domain Info table
- [ ] All analytics csv exports

Also verify that the analytics displays use converted fields.

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Update documentation in READMEs and/or onboarding guide

#### Ensured code standards are met (Original Developer)

- [ ] If any updated dependencies on Pipfile, also update dependencies in requirements.txt.
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] Tag @dotgov-designers in this PR's Reviewers for design review. If code is not user-facing, delete design reviewer checklist
- [ ] Verify new pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Verified code meets all checks above. Address any checks that are not satisfied
- [ ] Reviewed this code and left comments. Indicate if comments must be addressed before code is merged
- [ ] Checked that all code is adequately covered by tests
- [ ] Verify migrations are valid and do not conflict with existing migrations

#### Validated user-facing changes as a developer
**Note:** Multiple code reviewers can share the checklists above, a second reviewer should not make a duplicate checklist. All checks should be checked before approving, even those labeled N/A. 

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] (Rarely needed) Tested as both an analyst and applicant user

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior. Comment any found issues or broken flows.
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari
- [ ] (Rarely needed) Tested as both an analyst and applicant user

### References
- [Code review best practices](../docs/dev-practices/code_review.md)

### CODE REVIEW

File: src/registrar/admin.py [hunk 1]
```diff
@@ -2594,28 +2849,71 @@ def queryset(self, request, queryset):
...
+
+    # --- Generic Org Type
+    @admin.display(description=_("Converted Generic Org Type"))
+    def converted_generic_org_type(self, obj):
+        return obj.domain_info.converted_generic_org_type
+    converted_generic_org_type.admin_order_field = "domain_info__converted_generic_org_type"  # type: ignore
+    
     def generic_org_type(self, obj):
         return obj.domain_info.get_generic_org_type_display()
+    # generic_org_type.admin_order_field = "domain_info__generic_org_type"  # type: ignore
```
Comment (abroddrick) [inline src/registrar/admin.py:2862]: remove commented out code

File: src/registrar/admin.py [hunk 2]
```diff
@@ -2594,28 +2849,71 @@ def queryset(self, request, queryset):
...
+    def converted_federal_agency(self, obj):
+        return obj.domain_info.converted_federal_agency
+    converted_federal_agency.admin_order_field = "domain_info__converted_federal_agency"  # type: ignore
 
     def federal_agency(self, obj):
         if obj.domain_info:
             return obj.domain_info.federal_agency
         else:
             return None
+    # federal_agency.admin_order_field = "domain_info__federal_agency"  # type: ignore
```
Comment (abroddrick) [inline src/registrar/admin.py:2875]: remove commented out code

File: src/registrar/models/domain_request.py [hunk 3]
```diff
@@ -1416,8 +1416,8 @@ def _form_complete(self, request):
     @property
     def converted_organization_name(self):
         if self.portfolio:
-            return self.portfolio.organization_name
-        return self.organization_name
+            return "portfolio name" #self.portfolio.organization_name
```
Comment (abroddrick) [inline src/registrar/models/domain_request.py:1419]: **blocking**: looks like the values here should be the comment values

File: src/registrar/models/domain_information.py [hunk 4]
```diff
@@ -426,13 +426,14 @@ def get_state_display_of_domain(self):
...
             return None
 
+    # ----- Portfolio Properties -----
+
     @property
     def converted_organization_name(self):
         if self.portfolio:
-            return self.portfolio.organization_name
-        return self.organization_name
+            return "portoflio name" #self.portfolio.organization_name
```
Comment (abroddrick) [inline src/registrar/models/domain_information.py:434]: Looks like the value here should be the comments

File: src/registrar/admin.py [hunk 5]
```diff
@@ -2567,19 +2745,96 @@ def queryset(self, request, queryset):
...
+                return queryset.filter(
+                    # Filter based on the generic org value returned by converted_generic_org_type
+                    id__in=[
+                        domain.id for domain in queryset if domain.domain_info.converted_federal_type and domain.domain_info.converted_federal_type == self.value()
+                    ]
+                )
+            return queryset
+
+        
+    def get_queryset(self, request):
```
Comment (abroddrick) [inline src/registrar/admin.py:2809]: Why was this moved up? The typical convention I have seen is Class definition, followed by all class variables, then sub classes followed by class functions. We already have this format being broken by defining some variables above sub classes but I would prefer we not make the format even more outside of convention by moving functions up

File: src/registrar/admin.py [hunk 6]
```diff
@@ -2594,28 +2849,71 @@ def queryset(self, request, queryset):
         ),
     )
 
+    # ------- Domain Information Fields
+
+    # --- Generic Org Type
+    @admin.display(description=_("Converted Generic Org Type"))
```
Comment (abroddrick) [inline src/registrar/admin.py:2855]: **changes requested** @CocoByte we don't want this to show as "converted" anywhere for the users. This is not meant to be a user facing change and is meant to make sure users see the correct data and don't need to know how the sauce is made. Please change all the ones like this.

**Review by abroddrick — CHANGES_REQUESTED** [review]
See comments for changes requested and comments. Also I was only running this locally and already noticing a significant slowdown in performance. Can you please put this on a sandbox so we can see if the slow down is just a local oriented thing? If you want to use mine you can.

File: src/registrar/admin.py [hunk 7]
```diff
@@ -1453,25 +1454,110 @@ class DomainInformationResource(resources.ModelResource):
...
+    senior_official = fields.Field(attribute='converted_senior_official', column_name='senior_official')
+    address_line1 = fields.Field(attribute='converted_address_line1', column_name='address_line1')
+    address_line2 = fields.Field(attribute='converted_address_line2', column_name='address_line2')
+    city = fields.Field(attribute='converted_city', column_name='city')
+    state_territory = fields.Field(attribute='converted_state_territory', column_name='state_territory')
+    zipcode = fields.Field(attribute='converted_zipcode', column_name='zipcode')
+    urbanization = fields.Field(attribute='converted_urbanization', column_name='urbanization')
+
+    # Custom getters for the above columns that map to @property functions instead of fields
+    def dehydrate_organization_name(self, obj):
```
[the remark this answers is not in the document]
Comment (CocoByte) [reply src/registrar/admin.py:1473]: This is a convention used by django-import-export to override or customize the export behavior for specific fields.  Removing the "dehydrate_" keyword would require us to override the export_resource() method.

[the remark this answers is not in the document]
Comment (CocoByte) [reply src/registrar/admin.py:1473]: Without these methods, the converted fields will just get ignored.

File: src/registrar/admin.py [hunk 8]
```diff
@@ -1463,25 +1463,111 @@ class DomainInformationResource(resources.ModelResource):
     class Meta:
         model = models.DomainInformation
 
+    # Override exports for these columns in DomainInformation to use converted values.  These values
+    # come from @Property functions, which are not automatically included in the export and which we
+    # want to use in place of the native fields.
+    organization_name = fields.Field(attribute="converted_organization_name", column_name="organization_name")
```
[the remark this answers is not in the document]
Comment (abroddrick) [reply src/registrar/admin.py:1469]: @CocoByte have you tested the import/export fields? I would agree the Domain info Resource used for the import/export file doesn't need to be adjusted as that should just be purely what is in the database

File: src/registrar/tests/test_reports.py [hunk 9]
```diff
@@ -251,27 +251,31 @@ def test_domain_data_type(self):
...
         # sorted alphabetially by domain name
         expected_content = (
-            "Domain name,Status,First ready on,Expiration date,Domain type,Agency,Organization name,City,State,SO,"
-            "SO email,Security contact email,Domain managers,Invited domain managers\n"
-            "cdomain11.gov,Ready,2024-04-02,(blank),Federal - Executive,World War I Centennial Commission,,,,(blank),,,"
+            "Domain name,Status,First ready on,Expiration date,Domain type,Agency,"
+            "Organization name,City,State,SO,SO email,"
+            "Security contact email,Domain managers,Invited domain managers\n"
+            "adomain2.gov,Dns needed,(blank),(blank),Federal - Executive,"
+            "Portfolio 1 Federal Agency,,,, ,,(blank),"
```
[the remark this answers is not in the document]
Comment (CocoByte) [reply src/registrar/tests/test_reports.py:258]: Yes, I editted the data to make it clear when we are pulling from portfolio data or not.  adomain2 is affected across all these tests

File: src/registrar/utility/csv_export.py [hunk 10]
```diff
@@ -1085,20 +1231,6 @@ def get_filter_conditions(cls, **kwargs):
...
-        Get a dict of computed fields.
-        """
-        return {
-            "so_name": Concat(
-                Coalesce(F("senior_official__first_name"), Value("")),
-                Value(" "),
-                Coalesce(F("senior_official__last_name"), Value("")),
-                output_field=CharField(),
-            ),
-        }
```
[the remark this answers is not in the document]
Comment (CocoByte) [reply src/registrar/utility/csv_export.py:1100]: Yes.  so_name was replaced by converted_so_name

File: src/registrar/tests/test_reports.py [hunk 11]
```diff
@@ -721,11 +725,11 @@ def test_domain_request_growth(self):
             # spaces and leading/trailing whitespace
             csv_content = csv_content.replace(",,", "").replace(",", "").replace(" ", "").replace("\r\n", "\n").strip()
             expected_content = expected_content.replace(",,", "").replace(",", "").replace(" ", "").strip()
-
             self.assertEqual(csv_content, expected_content)
 
-    @less_console_noise_decorator
+    # @less_console_noise_decorator
     def test_domain_request_data_full(self):
+        self.maxDiff = None
```
[the remark this answers is not in the document]
Comment (CocoByte) [reply src/registrar/tests/test_reports.py]: We have these remnants in alot of places.  I'll clean them up

**Pull request merged**